src/utils/openShiftConnectorUtils.ts

```
@@ -0,0 +1,148 @@
+ */
+export function isOpenShiftConnectorInstalled(): boolean {
+  return !!extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID);
+}
+
+/**
+ * Returns the OpenShift Connector extension API
+ *
+ * @throws Error if the extension is not installed
+ * @returns the OpenShift Connector extension API
+ */
+export async function getOpenShiftConnector(): Promise<any> {
+  if (!isOpenShiftConnectorInstalled()) {
+    throw new Error(`${OPENSHIFT_CONNECTOR} is not installed`);
+  }
+  const openShiftConnector: Extension<any> = extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID);
+  if (openShiftConnector.isActive) {
+    return openShiftConnector.exports;
+  }
+  return extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID).activate();
+}
+
+/**
+ * Install the OpenShift Connector extension
+ *
+ * @returns when the extension is installed
+ * @throws if the user refuses to install the extension, or if the extension does not get installed within a timeout period
+ */
+async function installOpenShiftConnector(): Promise<void> {
+  const installListenerDisposables: Disposable[] = [];
+  return Promise.race([
+    new Promise<void>((resolve, reject) => {
+      extensions.onDidChange(() => {
+        if (isOpenShiftConnectorInstalled()) {
+          installListenerDisposables.forEach((d: Disposable) => { d.dispose(); });
+          resolve();
+        }
+      }, null, installListenerDisposables);
+      commands.executeCommand("workbench.extensions.installExtension", OPENSHIFT_CONNECTOR_EXTENSION_ID)
+        .then((_unused: any) => { }, reject);
+    }),
+    new Promise<void>((_unused, reject) => {
+      // Fail install if it takes longer than the timeout period
+      setTimeout(reject, DOWNLOAD_TIMEOUT, new Error(`${OPENSHIFT_CONNECTOR} installation is taking a while. Cancelling 'Deploy to OpenShift'`));
+    })
+  ]).catch(e => {
+    installListenerDisposables.forEach((d: Disposable) => { d.dispose(); });
+    throw e;
```

No need to re-throw the exception.